test/sslapitest.c

```diff
@@ -10985,6 +10985,60 @@ static int test_set_alpn(void)
     return testresult;
 }
 
+static int test_ec_point_formats(void)
+{
+    SSL_CTX *ctx = NULL;
+    SSL *ssl = NULL;
+    int testresult = 0;
+    unsigned char valid_formats[] = {
+        TLSEXT_ECPOINTFORMAT_uncompressed,
+        TLSEXT_ECPOINTFORMAT_ansiX962_compressed_prime,
+        TLSEXT_ECPOINTFORMAT_ansiX962_compressed_char2
+    };
+    unsigned char invalid_format[] = { TLSEXT_ECPOINTFORMAT_uncompressed, 0x99 };
+
+    ctx = SSL_CTX_new_ex(libctx, NULL, TLS_server_method());
+    if (!TEST_ptr(ctx))
+        goto end;
+
+    /* Test setting valid formats: */
+    if (!TEST_true(SSL_CTX_set_ec_point_formats(ctx, valid_formats, sizeof(valid_formats))))
+        goto end;
+
+    ssl = SSL_new(ctx);
+    if (!TEST_ptr(ssl))
+        goto end;
+
+    if (!TEST_true(SSL_set_ec_point_formats(ssl, valid_formats, sizeof(valid_formats))))
+        goto end;
+
+    /* Test setting invalid & empty formats: */
+    if (!TEST_false(SSL_CTX_set_ec_point_formats(ctx, invalid_format, sizeof(invalid_format))))
+        goto end;
+
+    if (!TEST_false(SSL_set_ec_point_formats(ssl, invalid_format, sizeof(invalid_format))))
+        goto end;
+
+    if (!TEST_false(SSL_CTX_set_ec_point_formats(ctx, NULL, 0)))
+        goto end;
+
+    if (!TEST_false(SSL_set_ec_point_formats(ssl, NULL, 0)))
+        goto end;
+
+    if (!TEST_false(SSL_CTX_set_ec_point_formats(ctx, valid_formats, 0)))
+        goto end;
+
+    if (!TEST_false(SSL_set_ec_point_formats(ssl, valid_formats, 0)))
+        goto end;
+
+    testresult = 1;
+
+end:
```

Comment:
I might not be a bad idea to run an openssl connection with a non-default ec point formats setting to confirm this completes successfully.